protocol/v1/status_message.go

```diff
@@ -328,7 +328,9 @@ func (m *StatusMessage) HandleApplication() error {
 		return m.unmarshalProtobufData(new(protobuf.SyncActivityCenterNotifications))
 	case protobuf.ApplicationMetadataMessage_SYNC_ACTIVITY_CENTER_NOTIFICATION_STATE:
 		return m.unmarshalProtobufData(new(protobuf.SyncActivityCenterNotificationState))
-	case protobuf.ApplicationMetadataMessage_COMMUNITY_ADMIN_MESSAGE:
+	case protobuf.ApplicationMetadataMessage_COMMUNITY_EVENTS_MESSAGE:
+		return m.unmarshalProtobufData(new(protobuf.CommunityEventsMessage))
+	case protobuf.ApplicationMetadataMessage_COMMUNITY_EVENTS_MESSAGE_REJECTED:
```

Comment:
Wondering whether we should have a different protobuf for this (or add some fields to the existing one)? 
I would add a `Reason` for the rejection for example (most of the cases would be that is out of date, but it could also be lack of permissions for example).
Either just a small wrapper, that way we don't have to have a nested decision in `handle...`, where we check the `type` of the metadata message in a nested decision,
or just adding a field to the existing protobuf.

What do you think?

Reply:
Updated, please take a second look.